### ISSUE TITLE
__get prefers model data over relationships

### ISSUE THREAD

**Issue by NiallSmyth** [RhubarbPHP/Module.Stem#23]
I'm not sure if this is something we should "fix", but there's a big gotcha when you've got data matching the name of a relationship in your model. The value in data will be returned, leaving you scratching your head about why you're not getting your relationship. This can happen when you're working with the data manually, or, perhaps more likely, when your model used to have a certain field and now it's been replaced by a relationship with the same name, but your existing database still has the old field.

**Comment by acuthbert** [comment]
Yeah it's a good point. I also hit an issue where a 'getter' function was added that superseeded a value in the model data. When filtering or sorting in unit tests it uses the getter, but when using the MySql repository it uses the table value so you can get different results.

Maybe we should not allow __get to replace any model values and throw some sort of exception when you setup that scenario so at least you get visibility of the issue?
